src/main/java/com/github/checkstyle/regression/module/UnitTestProcessorCheck.java

```diff
@@ -0,0 +1,211 @@
+////////////////////////////////////////////////////////////////////////////////
+
+package com.github.checkstyle.regression.module;
+
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import com.github.checkstyle.regression.data.ImmutableProperty;
+import com.github.checkstyle.regression.data.ModuleInfo;
+import com.puppycrawl.tools.checkstyle.DefaultConfiguration;
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+
+/**
+ * The custom check which processes the unit test class of a checkstyle module.
+ * This check would walk through the {@code @Test} annotation, find variable definition
+ * like {@code final DefaultConfiguration checkConfig = createModuleConfig(FooCheck.class)}
+ * and grab the property info from {@link DefaultConfiguration#addAttribute(String, String)}
+ * method call.
+ * @author LuoLiangchen
+ */
+public class UnitTestProcessorCheck extends AbstractCheck {
+    /** The map of unit test method name to properties. */
+    private final Map<String, Set<ModuleInfo.Property>> unitTestToProperties =
+            new LinkedHashMap<>();
+
+    @Override
+    public int[] getDefaultTokens() {
+        return new int[] {
+            TokenTypes.ANNOTATION,
+        };
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return new int[] {
+            TokenTypes.ANNOTATION,
+        };
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return new int[] {
+            TokenTypes.ANNOTATION,
+        };
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        if ("Test".equals(ast.findFirstToken(TokenTypes.IDENT).getText())) {
+            final DetailAST methodDef = ast.getParent().getParent();
+            final DetailAST methodBlock = methodDef.findFirstToken(TokenTypes.SLIST);
+            final Optional<String> configVariableName =
+                    getModuleConfigVariableName(methodBlock);
+            if (configVariableName.isPresent()) {
+                final Set<ModuleInfo.Property> properties = new LinkedHashSet<>();
+
+                for (DetailAST expr : getAllChildrenWithToken(methodBlock, TokenTypes.EXPR)) {
+                    if (isAddAttributeMethodCall(expr.getFirstChild(), configVariableName.get())) {
+                        final DetailAST elist =
+                                expr.getFirstChild().findFirstToken(TokenTypes.ELIST);
+                        final String key = convertExprToText(elist.getFirstChild());
+                        final String value = convertExprToText(elist.getLastChild());
+                        properties.add(ImmutableProperty.builder().name(key).value(value).build());
```

Comment:
It would have problems with non-string.

I viewed some test just now. I see smth like https://github.com/checkstyle/checkstyle/blob/master/src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/OperatorWrapCheckTest.java#L65.

Do you remember anything else that is non-string or `toString()`?

I would scan checkstyle repo and pay attention to this as well.